lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -799,6 +804,18 @@ public function getArrayResult()
         return $this->execute(null, self::HYDRATE_ARRAY);
     }
 
+    /**
+     * Gets the flat array of results for the query.
```

Comment:
The documentation for `getScalarResult` already mentions "flat". I'm myself not sure what is meant by that. Let's use something more precise? See 

https://github.com/doctrine/orm/blob/1de28c2cab5cd7e3c9d781a8beff22e328190911/docs/en/reference/dql-doctrine-query-language.rst#L1049

Please add documentation for the new hydration method .

```suggestion
     * Gets the one-dimensional array of results for the query.
```

Reply:
Updated